components/org.wso2.identity.apps.taglibs.layout.controller/src/main/java/org/wso2/identity/apps/taglibs/layout/controller/compiler/parsers/DefaultParser.java

```diff
@@ -0,0 +1,228 @@
+                    } else {
+                        condition = new ConditionIdentifier(currentIdentifierName,
+                                context.line.substring(context.start, context.matcher.start()));
+                    }
+                    context.start = context.matcher.end();
+                    ExecutableIdentifier child = compile(reader, context, currentIdentifierName);
+                    condition.setChild(child);
+                    allIdentifiers.add(condition);
+                } else if ((currentIdentifierName = context.matcher.group(4)) != null) {
+                    if (!identifierName.trim().equals(currentIdentifierName.trim())) {
+                        throw new IllegalStateException(
+                                "Layout file is not correctly written, " +
+                                        "make sure to place closing tags in correct place: " +
+                                        currentIdentifierName);
+                    }
+                    if (context.textKeeper != null) {
+                        allIdentifiers.add(new NoIdentifier(
+                                context.textKeeper + context.line.substring(context.start, context.matcher.start())));
+                        context.textKeeper = null;
+                    } else {
+                        allIdentifiers.add(
+                                new NoIdentifier(context.line.substring(context.start, context.matcher.start())));
+                    }
+                    context.start = context.matcher.end();
+                    return createCompiledObject(allIdentifiers);
+                } else if ((currentIdentifierName = context.matcher.group(5)) != null) {
+                    NotConditionIdentifier notCondition;
+                    if (context.textKeeper != null) {
+                        notCondition = new NotConditionIdentifier(currentIdentifierName,
+                                context.textKeeper + context.line.substring(context.start, context.matcher.start()));
+                        context.textKeeper = null;
+                    } else {
+                        notCondition = new NotConditionIdentifier(currentIdentifierName,
+                                context.line.substring(context.start, context.matcher.start()));
+                    }
+                    context.start = context.matcher.end();
+                    ExecutableIdentifier child = compile(reader, context, currentIdentifierName);
+                    notCondition.setChild(child);
+                    allIdentifiers.add(notCondition);
+                }
+            } else {
+                if (context.textKeeper == null) {
+                    context.textKeeper = new StringBuilder(context.line.substring(context.start) + "\n");
+                } else {
+                    context.textKeeper.append(context.line.substring(context.start) + "\n");
+                }
+                context.line = "";
```

Comment:
```suggestion
                context.line = StringUtils.EMPTY;
```
Use StringUtils.EMPTY instead of "" in other places in java code as well